unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/ViewAllComponentController.cs

```diff
@@ -0,0 +1,273 @@
+        INamesService namesService,
+        NotificationsController notificationsController)
+    {
+        this.view = view;
+        this.currentPlayerId = currentPlayerId;
+        this.wearablesCatalogService = wearablesCatalogService;
+        this.landsService = landsService;
+        this.namesService = namesService;
+        this.notificationsController = notificationsController;
+        view.OnBackFromViewAll += BackFromViewAll;
+        view.OnRequestCollectibleElements += RequestCollectibleElements;
+        view.OnClickBuyNft += (s1, s2) => OnClickBuyNft?.Invoke(s1, s2);
+    }
+
+    public void OpenViewAllSection(PassportSection section)
+    {
+        view.Initialize(section);
+    }
+
+    public void SetViewAllVisibility(bool isVisible)
+    {
+        view.SetVisible(isVisible);
+
+        if (isVisible)
+        {
+            sectionsCts = sectionsCts.SafeRestart();
+            cleanCachedWearablesPages = true;
+        }
+        else
+        {
+            sectionsCts.SafeCancelAndDispose();
+            sectionsCts = null;
+
+            view.SetLoadingActive(false);
+        }
+    }
+
+    private void BackFromViewAll()
+    {
+        OnBackFromViewAll?.Invoke();
+        SetViewAllVisibility(false);
+        view.CloseAllNftItemInfos();
+    }
+
+    private void ShowNftIcons(List<(NFTIconComponentModel model, WearableItem w)> wearables)
+    {
+        view.ShowNftIcons(wearables);
+    }
+
+    public void Dispose()
+    {
+        view.OnBackFromViewAll -= BackFromViewAll;
+        view.OnRequestCollectibleElements -= RequestCollectibleElements;
+
+        sectionsCts.SafeCancelAndDispose();
+        sectionsCts = null;
+    }
+
+    private void RequestCollectibleElements(PassportSection section, int pageNumber, int pageSize)
+    {
+        view.CloseAllNftItemInfos();
+
+        switch (section)
+        {
+            case PassportSection.Wearables:
+                RequestOwnedWearablesAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+            case PassportSection.Names:
+                RequestOwnedNamesAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+            case PassportSection.Lands:
+                RequestOwnedLandsAsync(currentPlayerId, pageNumber, pageSize, sectionsCts.Token).Forget();
+                break;
+        }
+    }
+
+    private async UniTask RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+    {
+        try
+        {
+            view.SetLoadingActive(true);
+
+            (IReadOnlyList<WearableItem> wearables, int totalAmount) ownedWearableItems =
+                await wearablesCatalogService.RequestOwnedWearablesAsync(userId, pageNumber, pageSize, cleanCachedWearablesPages, ct);
+
+            cleanCachedWearablesPages = false;
+            ProcessReceivedWearables(ownedWearableItems.wearables.ToArray());
```

Comment:
`ToArray` is redundant, you can use `IReadOnlyList` directly

Reply:
Totally agree! Fixed in: https://github.com/decentraland/unity-renderer/pull/4316/commits/f7f1425645c747cd69d451d08bd01d9f0a8a4afb